doc/whats-new.rst

```diff
@@ -21,9 +21,38 @@ v0.9.7 (unreleased)
 Enhancements
 ~~~~~~~~~~~~
 
+- Support for `pathlib.Path` objects added to
+  :py:func:`~xarray.open_dataset`, :py:func:`~xarray.open_mfdataset`,
+  :py:func:`~xarray.to_netcdf`, and :py:func:`~xarray.save_mfdataset`:
+
+  .. ipython::
+    :verbatim:
+    In [1]: import xarray as xr
+
+    In [2]: from pathlib import Path # In Python 2, use pathlib2!
+
+    In [3]: data_dir = Path("data/")
+
+    In [4]: one_file = data_dir / "dta_for_month_01.nc"
+
+    In [5]: print(xr.open_dataset(one_file))
+    Out[5]:
+    <xarray.Dataset>
+    [...]
+
+    In [6]: all_files = data_dir.glob("dta_for_month_*.nc")
```

Comment:
I removed this example from What's New for two reason:
1. The section was getting a little longer than essential
2. It's actually a bit of an anti-pattern, since the order of paths matters to xarray but is arbtirary from glob. The right way to write this is `sorted(data_dir.glob(...))`.